web/annotation_layer_builder.js

```diff
@@ -119,6 +119,24 @@ class AnnotationLayerBuilder {
       return;
     }
 
+    const uniqueLinks = linkAnnotations.filter(link => {
+      for (const annotation of annotations) {
+        const area = rect =>
+          Math.abs(rect[2] - rect[0]) * Math.abs(rect[3] - rect[1]);
+        const intersect = Util.intersect(annotation.rect, link.rect); // Find the intersection between the annotation and the link.
```

Comment:
This must also take `quadPoints` into account, when those exist, in addition to just the `rect`.

Reply:
This one turned out to be a bit tricky, also I was a bit unsure how to deal with comparing annotations with `quadPoints` and those without. It was also harder than I expected to find decent examples of multi-line links in PDFs in the wild but check out what I did and let me know how it looks.

Reply:
> I tried myself to build an example in either Word or with LaTeX and I didn't manage to have one: a splited url gives different annotations for each chunk.
> So it's likely possible to craft something (and I don't know how it'd supported in the different viewers) but I'm not sure it's worth spending too much time and that. So we could just add a TODO and wait for a bug.

@calixteman So you're suggesting that all quadPoints-handling can be removed from the patch (with a TODO comment about that)? If so, I suppose that's fine for now.

Reply:
Marking this as addressed since the lack of a test case is in a TODO comment now.

Reply:
This isn't fully resolved, since it's not clear (to me) if the suggestion was that we should remove all quadPoints handling or not!?

Reply:
For me it's fine to keep the intersection computation using quadPoints even if we don't have a test case.
And at some point, if we've a test case demonstrating that this code is wrong, then we'll fix and will add a test for it.